ambry-quota/src/main/java/com/github/ambry/quota/capacityunit/JsonCUQuotaSource.java

```diff
@@ -0,0 +1,311 @@
+    Map<String, CUQuota> quota = new HashMap<>();
+    ObjectMapper objectMapper = new ObjectMapper();
+    feQuota = objectMapper.readValue(config.frontendBandwidthCapacityInJson, CUQuota.class);
+    if (config.cuQuotaInJson != null && !config.cuQuotaInJson.trim().isEmpty()) {
+      Map<String, MapOrQuota> tempQuotas =
+          objectMapper.readValue(config.cuQuotaInJson, new TypeReference<Map<String, MapOrQuota>>() {
+          });
+      for (Map.Entry<String, MapOrQuota> entry : tempQuotas.entrySet()) {
+        Account account = accountService.getAccountById(Short.parseShort(entry.getKey()));
+        if (account == null) {
+          throw new IllegalStateException("No account id " + entry.getKey() + " is found in the account service");
+        }
+        if (account.getQuotaResourceType() == QuotaResourceType.ACCOUNT && !entry.getValue().isQuota
+            || account.getQuotaResourceType() != QuotaResourceType.ACCOUNT && entry.getValue().isQuota) {
+          throw new IllegalStateException(
+              "Account " + entry.getKey() + " quota enforcement type is different from account metadata: "
+                  + account.getQuotaResourceType());
+        }
+        if (entry.getValue().isQuota()) {
+          quota.put(QuotaResource.fromAccount(account).getResourceId(), entry.getValue().getQuota());
+        } else {
+          for (Map.Entry<String, CUQuota> containerQuotaEntry : entry.getValue().getContainerQuotas().entrySet()) {
+            String containerIdStr = containerQuotaEntry.getKey();
+            Container container = account.getContainerById(Short.parseShort(containerIdStr));
+            if (container == null) {
+              throw new IllegalStateException(
+                  "No container id " + containerIdStr + " is found in the account service under account "
+                      + entry.getKey());
+            }
+            quota.put(QuotaResource.fromContainer(container).getResourceId(), containerQuotaEntry.getValue());
+          }
+        }
+      }
+    }
+    this.cuQuota = quota;
+    this.cuUsage = new HashMap<>();
+    cuQuota.keySet().forEach(key -> cuUsage.put(key, new CUQuota(0, 0)));
+    feUsage = new CUQuota(0, 0);
+  }
+
+  @Override
+  public Quota<Long> getQuota(QuotaResource quotaResource, QuotaName quotaName) {
+    if (!SUPPORTED_QUOTA_NAMES.contains(quotaName)) {
+      return null;
+    }
+    if (!SUPPORTED_QUOTA_RESOURCE_TYPES.contains(quotaResource.getQuotaResourceType())) {
+      throw new IllegalArgumentException("Unsupported quota resource type: " + quotaResource.getQuotaResourceType());
+    }
+
+    String resourceId = quotaResource.getResourceId();
+    if (cuQuota.containsKey(resourceId)) {
+      return new Quota<>(quotaName, cuQuota.get(resourceId).getQuotaValue(quotaName), quotaResource);
+    }
+    return null;
+  }
+
+  @Override
+  public void updateNewQuotaResources(Collection<QuotaResource> quotaResources) {
```

Comment:
Is this guaranteed to be called only upon the creation of new accounts/containers?

Reply:
Its not. Fixed the logic to avoid updating the quota.